stdlib/importlib/metadata/__init__.pyi

```diff
@@ -123,7 +129,7 @@ class Distribution:
     @abc.abstractmethod
     def read_text(self, filename: str) -> str | None: ...
     @abc.abstractmethod
-    def locate_file(self, path: StrPath) -> PathLike[str]: ...
+    def locate_file(self, path: _TSimplePath) -> _TSimplePath: ...
```

Comment:
This would imply a guarantee that it returns the exact same type it was passed, which I don't think makes sense.

Reply:
That's true, the `path` should be the same type as passed to the constructor but there's no guarantee that the return value would be of the same type, e.g. if the distribution would extract the contents of a zip file.